File: LeetCode/Easy/Reshape the Matrix/Reshape the Matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        
        tmp = []
        for i in mat:
            for j in i:
                tmp.append(j)   
        if r*c != len(tmp):
            return mat
    
        index = 0
        answer = []
        
        for i in range(r):
            tmp_arry = []
            for j in range(c):
                tmp_arry.append(int(tmp[index]))
                index += 1
            answer.append(tmp_arry)
            
        return answer

File: LeetCode/Easy/Reshape the Matrix/Reshape the Matrix.go
func matrixReshape(mat [][]int, r int, c int) [][]int {
    ans := [][]int{}
    cnt := 0
    if len(mat)*len(mat[0])!=r*c{
        return mat
    }
    for _, row := range mat{
        for _, val := range row{
            
            if cnt%c == 0{
                ans = append(ans, []int{val})
            }else{
                ans[cnt/c] = append(ans[cnt/c], val)
            }
            cnt ++
        }
    }
    return ans
}
